crates/epoch-snark/src/epoch_block.rs

```diff
@@ -57,25 +54,28 @@ impl EpochBlock {
         Ok(expected_hash)
     }
 
-    pub fn blake2(&self) -> Result<Vec<bool>, EncodingError> {
-        Ok(hash_to_bits(&self.encode_to_bytes()?))
+    pub fn blake2(&self, encode_aggregated_pk: bool) -> Result<Vec<bool>, EncodingError> {
+        Ok(hash_to_bits(&self.encode_to_bytes(encode_aggregated_pk)?))
     }
 
     /// The goal of the validator diff encoding is to be a constant-size encoding so it would be
     /// more easily processable in SNARKs
-    pub fn encode_to_bits(&self) -> Result<Vec<bool>, EncodingError> {
+    pub fn encode_to_bits(&self, encode_aggregated_pk: bool) -> Result<Vec<bool>, EncodingError> {
```

Comment:
Can we get rid of this flag? It only seems to be set to `true` in tests, in order to create a failing test. We can create a failing test manually, no need to change our apis

Reply:
It's also used by a verifier.

Reply:
Will replace with explicit function call